specifyweb/frontend/js_src/lib/components/QueryBuilder/Header.tsx

```diff
@@ -55,31 +55,33 @@ export function QueryHeader({
   readonly toggleMapping: () => void;
 }): JSX.Element {
   return (
-    <header className="flex items-center gap-2 whitespace-nowrap">
-      <TableIcon label name={state.baseTableName} />
-      <H2 className="overflow-x-auto">
-        {typeof recordSet === 'object'
-          ? queryText.queryRecordSetTitle({
-              queryName: query.name,
-              recordSetTable: schema.models.RecordSet.label,
-              recordSetName: recordSet.get('name'),
-            })
-          : commonText.colonLine({
-              label: queryText.query(),
-              value: query.name,
-            })}
-      </H2>
-      {!queryResource.isNew() && <QueryEditButton query={query} />}
-      <span className="ml-2 flex-1" />
-      {!isScrolledTop && (
-        <Button.Small
-          onClick={(): void =>
-            form === null ? undefined : smoothScroll(form, 0)
-          }
-        >
-          {queryText.editQuery()}
-        </Button.Small>
-      )}
+    <header className="flex flex-col items-center justify-between gap-2 overflow-x-auto whitespace-nowrap sm:flex-row">
+      <div className="flex items-center justify-center gap-2">
+        <TableIcon label name={state.baseTableName} />
+        <H2 className="overflow-x-auto">
+          {typeof recordSet === 'object'
+            ? queryText.queryRecordSetTitle({
+                queryName: query.name,
+                recordSetTable: schema.models.RecordSet.label,
+                recordSetName: recordSet.get('name'),
+              })
+            : commonText.colonLine({
+                label: queryText.query(),
+                value: query.name,
+              })}
+        </H2>
+        {!queryResource.isNew() && <QueryEditButton query={query} />}
+        <span className="ml-2 flex-1" />
+        {!isScrolledTop && (
+          <Button.Small
+            onClick={(): void =>
+              form === null ? undefined : smoothScroll(form, 0)
+            }
+          >
+            {queryText.editQuery()}
+          </Button.Small>
+        )}
+      </div>
```

Comment:
These changes broke query header when querying on LoanPreparation table:

![image](https://user-images.githubusercontent.com/40512816/221258466-f60a3c03-64f0-4003-92b6-31a8167b694a.png)

Reply:
Also, there is a weird horizontal scroll bar in the header

Reply:
Scroll bar is gone, but button is still in a weird position:

<img width="1010" alt="Screenshot 2023-02-24 at 15 23 25" src="https://user-images.githubusercontent.com/40512816/221295788-065504f4-e29d-4281-8330-fae809d460cd.png">

Can you put it near the "Hide Field Mapper"?

Link: http://localhost/specify/query/new/loanpreparation/